lib/api/src/js/native.rs

```diff
@@ -64,11 +66,35 @@ macro_rules! impl_native_traits {
             pub fn call(&self, mut store: &mut impl AsStoreMut, $( $x: $x, )* ) -> Result<Rets, RuntimeError> where
             $( $x: FromToNativeWasmType + crate::js::NativeWasmTypeInto, )*
             {
-                let params_list: Vec<JsValue> = vec![ $( JsValue::from_f64($x.into_raw(&mut store))),* ];
-                let results = self.handle.get(store.as_store_ref().objects()).function.apply(
-                    &JsValue::UNDEFINED,
-                    &Array::from_iter(params_list.iter())
-                )?;
+                #[allow(unused_unsafe)]
+                let params_list: Vec<RawValue> = unsafe {
+                    vec![ $( RawValue { f64: $x.into_raw(store) } ),* ]
+                };
+                let params_list: Vec<JsValue> = params_list
+                    .into_iter()
+                    .map(|a| a.as_jsvalue(&store.as_store_ref()))
+                    .collect();
+                let results = {
+                    let mut r;
+                    //TODO: This loop is needed for async. It will be refactor with https://github.com/wasmerio/wasmer/issues/3451
```

Comment:
Can we add a space on the comment and change async to asyncify? (do this in all the comments)

```suggestion
// TODO: This loop is needed for asyncif. It will be refactored with https://github.com/wasmerio/wasmer/issues/3451
```

Reply:
done